tests/off/test_channels.py

```diff
@@ -27,300 +26,456 @@ def xfail_on_windows(func):
-    n_new_labels_2, _n_new_pulses_dict2 = data.refreshFromFiles()
-    assert n_new_labels_2 == 0
-
-
-def test_bad_channels_skipped():
-    # try:
-    data_local = ChannelGroup([filename])
-    assert len(data_local.keys()) == 1
-    ds_local = data_local.firstGoodChannel()
-    _, hists_pre_bad = data_local.hists(np.arange(10), "filtValue")
-    assert not ds_local.markedBadBool
-    ds_local.markBad("testing")
-    _, hists_post_bad = data_local.hists(np.arange(10), "filtValue")
-    assert len(hists_pre_bad) == 1
-    assert len(hists_post_bad) == 0
-    assert len(data_local.keys()) == 0
-    assert ds_local.markedBadBool
-    n_include_bad = 0
-    with data_local.includeBad():
-        for ds in data_local:
-            n_include_bad += 1
-    assert n_include_bad == 1
-    n_exclude_bad = 0
-    with data_local.includeBad(False):
-        for ds in data_local:
-            n_exclude_bad += 1
-    assert n_exclude_bad == 0
-
-
-def test_save_load_recipes(tmp_path):
-    data_local = ChannelGroup(getOffFileListFromOneFile(filename, maxChans=2))
-    ds_local = data_local.firstGoodChannel()
-    assert "energy" not in ds_local.__dict__, \
-        "ds_local should not have energy yet, we haven't defined that recipe"
-    pklfilename = str(tmp_path / "recipe_book_save_test2.rbpkl")
-    data.saveRecipeBooks(pklfilename)
-    ds = data.firstGoodChannel()
-    ds.add5LagRecipes(np.zeros(996))
-    data_local.loadRecipeBooks(pklfilename)
-    for ds in data.values():
-        ds_local = data_local[ds.channum]
-        assert all(ds.energy == ds_local.energy)
-
-
-def test_experiment_state_file_repeated_states():
-    # A better test would create an alternate experiment state file with repeated indicies and use that
-    # rather than reach into the internals of ExperimentStateFile
-    esf = mass.off.ExperimentStateFile(_parse=False)
-    # reach into the internals to simulate the results of parse with repeated states
-    esf.allLabels = ["A", "B", "A", "B", "IGNORE", "A"]
-    esf.unixnanos = np.arange(len(esf.allLabels)) * 100
-    esf.unaliasedLabels = esf.applyExcludesToLabels(esf.allLabels)
-    unixnanos = np.arange(2 * len(esf.allLabels)) * 50  # two entires per label
-    d = esf.calcStatesDict(unixnanos)
-    assert len(d["A"]) == esf.allLabels.count("A")
-    assert len(d["B"]) == esf.allLabels.count("B")
-    assert "IGNORE" not in d.keys()
-    for s in d["A"] + d["B"]:
-        assert s.stop - s.start == 2
+class TestOFFTutorial:  # noqa PLR0904
 
-    data_local = ChannelGroup([filename], experimentStateFile=esf)
-    ds_local = data_local.firstGoodChannel()
-    ds_local.stdDevResThreshold = 100
-    inds = ds_local.getStatesIndicies("A")
-    _ = ds_local.getAttr("filtValue", inds)
+    @pytest.mark.filterwarnings("ignore:invalid value encountered")
+    @pytest.mark.filterwarnings("ignore:divide by zero encountered")
+    @pytest.mark.filterwarnings("ignore:Polyfit may be poorly conditioned")
+    def setup_class(cls):
```

Comment:
Per https://docs.pytest.org/en/6.2.x/xunit_setup.html#class-level-setup-teardown `setup_class` is called once before any other class methods are called, and `setup_method` would be called before each method. So this is only called once and should be able to replicate the existing behavior.

I thought I'd answer my own question rather than just asking you.

Reply:
Yes. I did it with `setup_method` at first, and the tests took bloody forever. So I learned the answer to this empirically.